### PR TITLE
Additional offsets for Lineardeltakins

### PR DESCRIPTION
**Pull request by Ga-Ol-St**
Was using Machinekit for my custom build Rostock Linear Detla printer on BBB+CRAMPS(2.2.1)
as this device was build in home - it have hardware imperfections which Machinekit can't corrects with existing offsets in lineardeltakins - so i just add new Angle offsets and Radius offsets for Joint1 and Joint2, Joint0 - is always a reference point - so it don`t need offsets.

With new offsets i was abble to get very nice results with 6 factor callibration based on "least-squares calibration" available online http://www.escher3d.com/pages/wizards/wizarddelta.php

Sample CRAMPS configuration files which are using new offsets have modular design and based on python HAL generation. Also they are build for 4 degree of freedom (additional rotational axis) - but can be used without any changes on 3 Axis Delta - just don`t use A-axis - use Velocity Extrusion method to control extrusion.

Please review and give some feedback.
Thanks.

### CODE REVIEW

**Comment by ArcEye** [comment]
Author not whitelisted yet - can one of the admins verify this patch?

**Comment by luminize** [comment]
Add to whitelist

**Comment by ArcEye** [comment]
@luminize  I was unaware of this one

Are you able to review and merge if OK?

**Comment by luminize** [comment]
Sure, i'll have a look

File: configs/ARM/BeagleBone/Rostock-4DOF-CRAMPS/delta.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import os
+
+from machinekit import hal
+#from machinekit import rtapi as rt
```
Comment (luminize) [inline configs/ARM/BeagleBone/Rostock-4DOF-CRAMPS/delta.py]: could you remove this commented line?

File: configs/ARM/BeagleBone/Rostock-4DOF-CRAMPS/rostock-3dof.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+# Extruder, velocity controlled
+for i in range(0, numExtruders):
+    base.setup_stepper(section='EXTRUDER_%i' % i, stepgenIndex=3,
+                       axisIndex=(i+3), thread='servo-thread')
+
+# Extruder Multiplexer
+base.setup_extruder_multiplexer(extruders=numExtruders, thread='servo-thread')
+
+# Stepper Multiplexer
```
Comment (luminize) [inline configs/ARM/BeagleBone/Rostock-4DOF-CRAMPS/rostock-3dof.py]: following lines are commented out. Is the functionality still around when uncommented? If not, can you remove the commented lines?

File: configs/ARM/BeagleBone/Rostock-4DOF-CRAMPS/run.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/python
+
+import sys
+import os
+import subprocess
+import time
+from machinekit import launcher
+
+launcher.register_exit_handler()
+#launcher.set_debug_level(5)
```
Comment (luminize) [inline configs/ARM/BeagleBone/Rostock-4DOF-CRAMPS/run.py]: same comment about comments goes for this line, and line 17

File: src/emc/kinematics/lineardeltakins.c [hunk 4]
```diff
@@ -94,13 +94,33 @@ int rtapi_app_main(void)
         retval = hal_pin_float_newf(HAL_IN, &haldata->j2off, comp_id,
                 "lineardeltakins.J2off");
 
+    if(retval == 0)
```
[the remark this answers is not in the document]
Comment (Ga-Ol-St) [reply src/emc/kinematics/lineardeltakins.c]: Done.

File: configs/ARM/BeagleBone/Rostock-4DOF-CRAMPS/run.py (hunk 3 above)
Comment (Ga-Ol-St) [reply configs/ARM/BeagleBone/Rostock-4DOF-CRAMPS/run.py]: removed.

File: configs/ARM/BeagleBone/Rostock-4DOF-CRAMPS/delta.py (hunk 1 above)
Comment (Ga-Ol-St) [reply configs/ARM/BeagleBone/Rostock-4DOF-CRAMPS/delta.py]: done.

**Comment by luminize** [comment]
thanks @Ga-Ol-St

File: configs/ARM/BeagleBone/Rostock-4DOF-CRAMPS/rostock-3dof.py (hunk 2 above)
Comment (Ga-Ol-St) [reply configs/ARM/BeagleBone/Rostock-4DOF-CRAMPS/rostock-3dof.py]: Basically this file should be in sync with
configs/ARM/BeagleBone/MendelMax-CRAMPS/mendelmax.py 
i just added lineardelta configuration

fixed.

**Pull request merged**